eventstore/manager.go

```diff
@@ -0,0 +1,108 @@
+package eventstore
+
+import (
+	"io"
+
+	"github.com/google/uuid"
+	ds "github.com/ipfs/go-datastore"
+	kt "github.com/ipfs/go-datastore/keytransform"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/textileio/go-textile-core/threadservice"
+)
+
+var (
+	dsStoreManagerBaseKey = ds.NewKey("/manager")
+)
+
+type Manager struct {
+	io.Closer
+
+	config *StoreConfig
+
+	threadservice threadservice.Threadservice
```

Comment:
The manager needs a threadservice but it does _not_ deal with closing it. In practice, the daemon uses two datastores: one for the manager and one for the thread service.